receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -295,6 +298,12 @@ func (t *transaction) AppendHistogram(_ storage.SeriesRef, ls labels.Labels, atM
 	default:
 	}
 
+	var schema int32
+	if h != nil {
+		schema = h.Schema
+	} else if fh != nil {
+		schema = fh.Schema
+	}
 	t.addingNativeHistogram = true
```

Comment:
This is the place I had in mind in the comment above. Couldn't we add a second flag here, e.g.

```go
	var schema int32
	if h != nil {
		schema = h.Schema
	} else if fh != nil {
		schema = fh.Schema
	}
	t.addingNativeHistogram = true
	if schema == -53 {
		t.addingNHCB = true
	}
```

If we use that flag when calling getOrCreateMetricFamily, then the schema argument isn't needed and we don't need to add -1 in weird places 😅